inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,220 @@
+
+        if hits:
             try:
-                data['page_nr'] = int(page_end) - int(page_start) + 1
-            except (TypeError, ValueError):
+                form_fields['journal_title'] = hits[0].short_titles[0].title
+            except (AttributeError, IndexError):
                 pass
 
-    # ============================
-    # Language
-    # ============================
-    if form_fields.get('language') == 'oth':
-        if form_fields.get("other_language"):
-            data["languages"] = [form_fields["other_language"]]
-
-    # ==========
-    # Owner Info
-    # ==========
-    # TODO Make sure we are getting the email correctly
-    userid = obj.id_user
+    # journal location
+    page_range = form_fields.get('page_range_article_id')
+
+    artid = None
+    page_end = None
+    page_start = None
+    if page_range:
+        page_start, page_end, artid = split_page_artid(page_range)
+
+    builder.add_publication_info(
+        year=year,
+        cnum=form_fields.get('conference_id'),
+        journal_issue=form_fields.get('issue'),
+        journal_title=form_fields.get('journal_title'),
+        journal_volume=form_fields.get('volume'),
+        page_start=page_start,
+        page_end=page_end,
+        artid=artid
+    )
+
+    if form_fields.get('preprint_created'):
+        builder.add_imprint_date(form_fields.get('preprint_created'))
+
+    builder.add_thesis(
+        defense_date=form_fields.get('defense_date'),
+        degree_type=form_fields.get('degree_type'),
+        institution=form_fields.get('institution'),
+        date=form_fields.get('thesis_date')
+    )
+    if form_fields.get('experiment'):
+        builder.add_accelerator_experiment(experiment=form_fields.get('experiment'))
+
+    language = _language_mapping(
+        form_fields.get('language'),
+        form_fields.get('other_language')
+    )
+    if language and len(language) == 2:
+        builder.add_language(language=language)
+
+    if form_fields.get('title_translation'):
+        builder.add_title_translation(
+            title=form_fields.get('title_translation'),
+            language=language
+        )
+
+    if form_fields.get('title_arXiv'):
+        builder.add_title(
+            title=form_fields.get('title_arXiv'),
+            source='arXiv'
+        )
+
+    if form_fields.get('title_crossref'):
+        builder.add_title(
+            title=form_fields.get('title_crossref'),
+            source='crossref'
+        )
+
+    if form_fields.get('license_url'):
+        builder.add_license_url(license_url=form_fields.get('license_url'))
+
+    if form_fields.get('public_notes'):
+        builder.add_public_note(form_fields.get('public_notes'))
+
+    if form_fields.get('note'):
+        builder.add_public_note(
+            form_fields.get('note'),
+            source='arXiv' if form_fields.get('categories') else 'CrossRef'
+        )
```

Comment:
I am confused here: @jmartinm are arXiv notes and Crossref notes not going into `public_notes`? Is the user able to submit some public notes? I have also seen above plenty ways of hidden notes...